docs/proposals/introducing_relaxed_data_binding.md

```diff
@@ -0,0 +1,98 @@
+
+## Goals
+- Enhance the Ballerina HTTP module by adding support to handle undefined nillable properties where the users can choose strict data binding or relaxed data binding.
+
+## Motivation
+According to OpenAPI specification if a given property is required or/and nullable we need to specifically mention it as explained below.
+
+If a given property is `required`,
+```yaml
+type: object
+properties:
+  id:
+    type: integer
+  username:
+    type: string
+required:
+  - id
+  - username
+```
+
+If a given property is `nullable`,
+```yaml
+type: integer
+nullable: true
+```
+
+Therefore if a given property is not specifically mentioned as required, then that property is optional by default. Similarly if a given property is not specifically mentioned as `nullable: true`, then that property is not accepting null values by default.
+
+There are several scenarios where the received response has null values even though that property is not configured as `nullable: true`. In such cases http target type binding fails, returning a type conversion error.
+
+As a workaround, there is a `--nullable` flag to make all the fields nillable when it is not defined in the OpenAPI specification. However this approach also makes all the fields nullable which is not an ideal solution.
+
+A more user-friendly solution is to implement relaxed data binding, which would allow the tool to accept null values for non-nullable fields without raising runtime errors. This approach would handle unexpected null values gracefully, reducing type conversion issues and enhancing the developer experience by aligning more closely with API responses.
+
+## Description
+Following is the breakdown of the cases we need to reconsider within the context of payload data binding:
+<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-f6c8a312-7fff-fcbf-faf3-0b5d1af4efa6"><div dir="ltr" style="margin-left:0pt;" align="left">
+Scenario | Ballerina Representation | Current Behavior | Expected Behavior
+-- | -- | -- | --
+Required & non-nullable | T foo; | Both null values and absent fields cause runtime failures | Same
+Required & nullable | T? foo; | Absent fields will cause runtime failures | Need relaxed data binding to map absent fields as nil values
+Optional & non-nullable | T foo?; | Null values will cause runtime failures | Need relaxed data binding to map null values as absent fields
+Optional & nullable | T? foo?; | Both nil values and absent fields are allowed | Same
+
+</div></b>
+
+In the `data.jsondata` package, the `Options` configuration already provides support for handling various field types in JSON data binding, accommodating scenarios with both absent fields and nil values. The `Options` record allows us to adjust how nullability and optionality are treated, making it a versatile solution for managing the differences between required and optional fields, as well as nullable and non-nullable fields.
```

Comment:
```suggestion
In the `data.jsondata` package, the `Options` configuration already provides support for handling various field types in JSON data binding, accommodating scenarios with both absent fields and null values. The `Options` record allows us to adjust how nullability and optionality are treated, making it a versatile solution for managing the differences between required and optional fields, as well as nullable and non-nullable fields.
```

Lets use either one of this terminology: `null` or `nil`. Check other places and make it consistence